_compdemos/Cromwell.md

```diff
@@ -4,18 +4,101 @@ primary_reviewers: vortexing
+
+### Software and Environments
+One challenge associated with reproducibility and portability that is something often undervalued in many bioinformatic analyses, is the management of computing software and environments.  As projects evolve, they often begin in a testing phase with small datasets using readily available software or packages on a desktop computer or even potentially on a shared, remote computing resource. As they scale, it often is natural to simply do X, but Y times, all using the same computing environment of the user who originally developed the workflow.  
+
+While this is common, it also creates headaches later.  Using a workflow manager like Cromwell as tool for deploying your analyses means also that you can begin your work directly with developing a WDL and running the tests via Cromwell.  Then as you develop your workflow and want to scale it to an entire analysis, you can simply edit your workflow and Cromwell will keep track of all the tasks and their interrelations.  
+
+While workflows are being developed in the context of a WDL for running with Cromwell (or other WDL runners that are being developed), each task must be defined to occur in a specific environment.  That could be a user's environment on the local SLURM cluster, or that could be a clean environment with specific software packages loaded, or it could be a docker or singularity container that directly specifies the software and environment in which to run the task. 
+
+#### EasyBuild and Lmod
+One approach to making software versioning and environment management easier and available to users on a shared HPC cluster like our SLURM cluster, is the use of a "module" system.  We have more information in this Wiki on [managing your software environment at the Hutch](/scicomputing/compute_environments/), but to summarize here, essentially two users can load a single, defined module when they want to run a task and the versions of all the software required for that module to work will be loaded identically for either user.  That means that regardless of who is running a task, all users can load the same version of a software package to ensure that they get the same results!  For those of us here at the Hutch who have access to a [wide variety of software modules](/scicomputing/compute_scientificSoftware/) (built via [EasyBuild](https://github.com/FredHutch/easybuild-life-sciences) for us!), this is a useful and time-saving system that can very easily be leveraged using our configuration of Cromwell. 
```

Comment:
One approach to making software versioning and environment management easier, and accessible to users on a shared HPC cluster like our SLURM cluster, is the use of a "module" system.